APCAppCore/APCAppCoreTests/Reminder Tests/APCTasksReminderManagerTests.m

```diff
@@ -0,0 +1,240 @@
+- (void) setUp
+{
+    [super setUp];
+    
+    self.reminderManager = [[MockAPCTasksReminderManager alloc] init];
+    
+    self.reminderManager.updatingRemindersRemovesAllLocalNotifications = YES;
+    
+    self.reminderManager.reminderTime = [APCTasksReminderManager reminderTimesArray][17]; // 17 = 5pm
+    
+    [self.reminderManager setReminderMessage:@"Reminder Message"
+                             andDelayMessage:@"Delay Reminder 1 Hour"];
+    
+    [self.reminderManager handleActivitiesUpdateWithTodaysTaskGroups:@[]];
+    
+    APCTaskReminder *walkingAssessmentReminder = [[APCTaskReminder alloc] initWithTaskID:kWalkingTaskReminder reminderBody:kWalkingName];
+    APCTaskReminder *cardiacHealthReminder = [[APCTaskReminder alloc] initWithTaskID:kCardiacTaskReminder reminderBody:kCardiacName];
+    APCTaskReminder *balanceAssessmentReminder = [[APCTaskReminder alloc] initWithTaskID:kBalanceTaskReminder reminderBody:kBalanceName];
+    
+    [self.reminderManager setReminderKey:kWalkingTaskReminder toOn:YES];
+    [self.reminderManager setReminderKey:kCardiacTaskReminder toOn:YES];
+    [self.reminderManager setReminderKey:kBalanceTaskReminder toOn:YES];
+    
+    [self.reminderManager setAllReminders:YES];
+    
+    self.reminderManager.tasksFullComplete = @{
+                                               kWalkingTaskReminder : @(NO),
+                                               kCardiacTaskReminder : @(NO),
+                                               kBalanceTaskReminder : @(NO)};
+    
+    [self.reminderManager.reminders removeAllObjects];
+    [self.reminderManager manageTaskReminder:walkingAssessmentReminder];
+    [self.reminderManager manageTaskReminder:cardiacHealthReminder];
+    [self.reminderManager manageTaskReminder:balanceAssessmentReminder];
+}
+
+/** Put teardown code here. This method is called after the invocation of each test method in the class. */
+- (void) tearDown
+{
+    [super tearDown];
+}
+
+- (void) testRemindersTurnedOff
+{
+    [self.reminderManager setAllReminders:NO];
+    [self.reminderManager updateTasksReminder];
+    XCTAssertEqual(0, self.reminderManager.scheduledLocalNotification.count);
+}
+
+- (void) testFphsSundayTuesdayNoneCompleteOnSaturday
+{
+    // Saturday
+    self.reminderManager.mockNow = [NSDate dateWithISO8601String:@"2016-01-23T10:00:00+00:00"];
+    
+    self.reminderManager.daysOfTheWeekToRepeat = @[@(kAPCTaskReminderDayOfWeekSunday),
+                                                   @(kAPCTaskReminderDayOfWeekTuesday)];
+    
+    [self.reminderManager updateTasksReminder];
+    
+    XCTAssertEqual(2, self.reminderManager.scheduledLocalNotification.count);
+    
+    UILocalNotification* sundayAt5Pm = self.reminderManager.scheduledLocalNotification[0];
+    UILocalNotification* tuesdayAt5Pm = self.reminderManager.scheduledLocalNotification[1];
+    
+    NSDate* nextSundayAt5PM = [NSDate dateWithISO8601String:@"2016-01-24T22:00:00+00:00"];
```

Comment:
Sunday at 5pm where?  In Greenwich, it's 10PM, which means that in Seattle it's 2PM and in Pittsburgh it is 5PM. Will this test will fail if you are in a different timezone?